src/QmlControls/FactValueGrid.cc

```diff
@@ -96,6 +102,14 @@ void FactValueGrid::setFontSize(FontSize fontSize)
     }
 }
 
+void FactValueGrid::setVehicle(Vehicle * vehicle)
+{
+    _vehicle = vehicle;
```

Comment:
And then given the `REQUIRED` property I would change this to two things:
* qCritical if incoming vehicle is null
* And then qCritical if called when _vehicle is not null which indicates multiple calls. And then you don't need the _componentCompleted flag.